### PR TITLE
Feat/links

### PR DESCRIPTION
**Pull request by freesteph**
Première partie de la réécriture des liens 🎉

### CODE REVIEW

File: app/helpers/dsfr_link_helper.rb [hunk 1]
```diff
@@ -1,113 +1,51 @@
...
-  end
-
-  def dsfr_button_classes(*styles, default_class: 'fr-btn')
-    if (invalid_styles = (styles - BUTTON_STYLES.keys)) && invalid_styles.any?
-      fail(ArgumentError, "invalid styles #{invalid_styles.to_sentence}. Valid styles are #{BUTTON_STYLES.keys.to_sentence}")
-    end
-
-    [default_class] + BUTTON_STYLES.values_at(*styles).compact
-  end
+  ICON_SIDES = %w(left right).freeze
```
Comment (adipasquale) [inline app/helpers/dsfr_link_helper.rb:2]: I'm guessing `position` would be a better name, wdyt?

File: guide/content/helpers/link.slim [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  code: all_links) do
+  markdown:
+    Quelques combinaisons (au fil du texte/interne,
+    hors-texte/externe, etc.) sont exposées ici.
+
+== render('/partials/example.*',
+  caption: "Lien au fil du texte",
+  code: link_embed) do
+  markdown:
+    Liens sans attributes particuliers, au fil du texte. Ils n'ont pas
```
Comment (adipasquale) [inline guide/content/helpers/link.slim:38]: attributs

File: guide/layouts/partials/related-info.slim [hunk 3]
```diff
@@ -3,4 +3,4 @@ h2
 
 ul
   - links.each do |text, link|
-    li== link_to text, link
+    li== link_to text, link, {target: '_blank', rel: 'noopener'}
```
Comment (adipasquale) [inline guide/layouts/partials/related-info.slim:6]: 👍

Comment (adipasquale) [reply guide/layouts/partials/related-info.slim:6]: nitpick: curly brackets are optional here

**Review by adipasquale — COMMENTED** [review]
👍 well done I think it's quite simple now, both to use and to read the code 

rubocop is not happy with a few things, I think it's confusing `href: #` for a comment start, and other things. 
There might still be something mysterious in its config if you don't have the same behaviour locally.

optionals : 

1. could you add a make command or some doc somewhere on how to use `guard` ? It's been a looong while i  haven't worked with it
2. It would be nice to add some yard doc on the accepted keyword arg options for dsfr_link_to if you're up for it 
3. Icons don't show up in the doc but that's independent from your PR, we'll have to fix it
4.  in slim guide templates `==` does not escape html whereas `=` does. But for html_safe helpers such as dsfr_link_to / link_to it's enough to use `=` (or I'm missing something)

**Review by adipasquale — APPROVED** [review]

File: app/helpers/dsfr_link_helper.rb (hunk 1 above)
Comment (freesteph) [reply app/helpers/dsfr_link_helper.rb:2]: agreed

**Comment by freesteph** [comment]
> 👍 well done I think it's quite simple now, both to use and to read the code

Thanks!

> 
> rubocop is not happy with a few things, I think it's confusing `href: #` for a comment start, and other things. There might still be something mysterious in its config if you don't have the same behaviour locally.

I've fixed Rubocop and it didn't mention `href`, strange.

By the way I'm trialling a new Git approach where I do a bunch of fixup commits so we don't loose our diff, and then once we're both happy with it I can interactive-rebase and merge.
> 
> optionals :
> 
>     1. could you add a make command or some doc somewhere on how to use `guard` ? It's been a looong while i  haven't worked with it

done

> 
>     2. It would be nice to add some yard doc on the accepted keyword arg options for dsfr_link_to if you're up for it
> 
Yep, sorry forgot about this bit. Done now.

>     3. Icons don't show up in the doc but that's independent from your PR, we'll have to fix it

Right so this is an absolute pain in the neck but basically the DSFR looks up icons at two different paths which makes it incredibly tricky to serve. I am working on a fix but let's save it for another day.
> 
>     4. in slim guide templates `==` does not escape html whereas `=` does. But for html_safe helpers such as dsfr_link_to / link_to it's enough to use `=` (or I'm missing something)

Fixed it thanks.

If you want to have another look, I'm all good.

**Review by adipasquale — APPROVED** [review]
🚀 well done thanks for the quick fixes 🙇

**Pull request merged**